File: configmodels/config_msg.go
// SPDX-FileCopyrightText: 2021 Open Networking Foundation <info@opennetworking.org>
//
// SPDX-License-Identifier: Apache-2.0
//

package configmodels

import (
	"github.com/omec-project/openapi/models"
)

const (
	Post_op = iota
	Put_op
	Delete_op
)

const (
	Device_group = iota
	Network_slice
	Sub_data
)

type ConfigMessage struct {
	DevGroup     *DeviceGroups
	Slice        *Slice
	AuthSubData  *models.AuthenticationSubscription
	DevGroupName string
	SliceName    string
	Imsi         string
	MsgType      int
	MsgMethod    int
}

// Slice + attached device group
type SliceConfigSnapshot struct {
	SliceMsg *Slice
	DevGroup []*DeviceGroups
}

// DevGroup + slice name
type DevGroupConfigSnapshot struct {
	DevGroup  *DeviceGroups
	SliceName string
}

File: ui/frontend_files/modules/objectsModels/config_msg.js
/**
 * JavaScript equivalent of Go structs from config_msg.go
 */

import { DeviceGroups } from "./model_device_groups";
import { Slice } from "./model_slice";

// Constants matching Go enums
export const OperationType = {
  POST: 0,
  PUT: 1,
  DELETE: 2
};

export const GroupType = {
  DEVICE_GROUP: 0,
  NETWORK_SLICE: 1,
  SUB_DATA: 2
};

/**
 * ConfigMessage represents configuration messages for different entities
 */
export class ConfigMessage {
  constructor() {
    /** @type {DeviceGroups|null} */
    this.DevGroup = null;       // DeviceGroups
    /** @type {Slice|null} */
    this.Slice = null;          // Slice
    // TODO: implement the object *models.AuthenticationSubscription
    this.AuthSubData = null;    // AuthenticationSubscription
    this.DevGroupName = '';
    this.SliceName = '';
    this.Imsi = '';
    this.MsgType = 0;
    this.MsgMethod = 0;
  }
}

/**
 * Represents a slice with its attached device groups
 */
export class SliceConfigSnapshot {
  constructor() {
    /** @type {Slice|null} */
    this.SliceMsg = null;      // Slice
    /** @type {Array<DeviceGroups>|null} */
    this.DevGroup = [];        // Array of DeviceGroups
  }
}

/**
 * Represents a device group with its slice name
 */
export class DevGroupConfigSnapshot {
  constructor() {
    /** @type {DeviceGroups|null} */
    this.DevGroup = null;      // DeviceGroups
    this.SliceName = '';
  }
}
